stan/math/prim/mat/fun/matrix_exp_pade.hpp

```diff
@@ -6,34 +6,34 @@
 namespace stan {
   namespace math {
 
-        /**
-         * Computes the matrix exponential, using a Pade
-         * approximation, coupled with scaling and
-         * squaring.
-         *
-         * @tparam MatrixType scalar type of the elements
-         * in the input matrix.
-         * @param[in] arg matrix to exponentiate.
-         * @param[out] Matrix exponential of input matrix.
-         */
-        template <typename MatrixType>
-        MatrixType
-        matrix_exp_pade(const MatrixType& arg) {
-          MatrixType result;
-          MatrixType U, V;
-          int squarings;
-          Eigen::matrix_exp_computeUV<MatrixType>::run(
-            arg, U, V, squarings, arg(0, 0));  // Pade approximant is
-                                               // (U+V) / (-U+V)
-          MatrixType numer = U + V;
-          MatrixType denom = -U + V;
-          result = denom.partialPivLu().solve(numer);
-          for (int i = 0; i < squarings; ++i)
-            result *= result;  // undo scaling by repeated squaring
+    /**
+     * Computes the matrix exponential, using a Pade
+     * approximation, coupled with scaling and
+     * squaring.
+     *
+     * @tparam MatrixType scalar type of the elements
+     * in the input matrix.
+     * @param[in] arg matrix to exponentiate.
+     * @return Matrix exponential of input matrix.
+     */
+    template <typename MatrixType>
+    MatrixType
+    matrix_exp_pade(const MatrixType& arg) {
+      MatrixType result;
```

Comment:
I didn't catch this the first time, but I really don't like this style of declaring a variable `result` way up here and then not using it until line 31.  It should be declared and defined both on line 31.  It also saves constructing a matrix that's never used (the nullary constructor gets called in the first declaration without a definition, though `Eigen::Matrix` is minimal compared to `std::vector` in its default constructor).